app/Services/Email.php

```diff
@@ -138,25 +138,36 @@ protected function setupEmail()
             $this->allMessages[$key]['user'] = $user;
 
             $tokens = $this->defineTokens($user);
-            $this->allMessages[$key]['message'] = $this->processMessage($tokens, $this->message);
 
-            if ($this->message->type == 'leaderboard') {
-                $this->allMessages[$key]['message']['leaderboard'] = $this->generateLeaderboard($this->users);
+            $message = [];
+
+            $message = $this->processMessage($tokens, $this->message);
 
-                // @TODO - get top 3 reportbacks
+            if ($this->message->type == 'leaderboard') {
+                $message = array_merge($message, $this->getLeaderboardVars($this->users));
```

Comment:
does this need to be processed within the `foreach`? it's the same for every user, right?